app/cdap/components/Administration/TetheringTabContent/NewTetheringRequest/index.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import { Link, useHistory } from 'react-router-dom';
+import T from 'i18n-react';
+import { TetheringApi } from 'api/tethering';
+import { IApiError } from '../types';
+import { IErrorObj } from 'components/shared/ConfigurationGroup/utilities';
+import Alert from 'components/shared/Alert';
+import styled from 'styled-components';
+import OmniNamespaces from './OmniNamespaces';
+import CdfInfo from './CdfInfo';
+import { HeaderContainer, HeaderTitle, BodyContainer, StyledButton } from '../shared.styles';
+
+const I18NPREFIX = 'features.Administration.Tethering.CreateRequest';
+
+const Container = styled.div`
+  width: 100vw;
+  margin-top: -20px;
+  margin-left: calc(-50vw + 50%);
+`;
+
+const BackButton = styled(Link)`
+  margin-top: 1px;
+  font-size: 1rem;
+
+  &:hover {
+    text-decoration: none;
+  }
+`;
+
+const Divider = styled.span`
+  font-size: 1.25rem;
+  padding: 0 5px;
+`;
+
+const StyledBodyContainer = styled(BodyContainer)`
+  padding: 30px;
+  flex-direction: column;
+  align-items: flex-start;
+`;
+
+const ButtonsContainer = styled.div`
+  margin-left: -30px;
+`;
+
+interface IValidationErrors {
+  instanceName?: IErrorObj;
+  projectName?: IErrorObj;
+  region?: IErrorObj;
+}
+
+const NewTetheringRequest = () => {
+  const [namespace, setNamespace] = useState('');
+  const [cpuLimit, setCpuLimit] = useState('');
+  const [memoryLimit, setMemoryLimit] = useState('');
+  const [projectName, setProjectName] = useState('');
+  const [region, setRegion] = useState('');
+  const [instanceName, setInstanceName] = useState('');
+  const [showAlert, setShowAlert] = useState(false);
+  const [apiErr, setApiErr] = useState({} as IApiError);
+  const [validationErrors, setValidationErrors] = useState({} as IValidationErrors);
+  const history = useHistory();
+
+  const stateUpdater = {
+    name: setNamespace,
+    cpuLimit: setCpuLimit,
+    memoryLimit: setMemoryLimit,
+    projectName: setProjectName,
+    region: setRegion,
+    instanceName: setInstanceName,
+  };
+
+  const resetValues = () => {
+    setNamespace('');
```

Comment:
Series of state updates will trigger multiple re-renders i guess. You could consider moving these into a single object and update it once.

const [userState, setUserState] = useState({
  namespace: '',
  cpuLimit: '',
  ......
});